Assets/Scripts/CameraController.cs

```diff
@@ -59,12 +68,30 @@ private void Update()
             _isMouseRotating = false;
 
         if (_isKeyboardRotating)
+        {
             RotateKeyboardCamera();
+        }
 
         if (_isMouseRotating)
         {
             RotateMouseCamera();
         }
+
+        UpdateFadingBlocks();
+    }
+
+    private void UpdateFadingBlocks()
+    {
+        Vector3 cameraPosition = transform.position;
+        Vector3 playerPosition = PlayerObject.transform.position;
+        foreach (BlockBehaviour blockBehavior in _wallBlocks)
+        {
+            if ((cameraPosition - blockBehavior.transform.position).magnitude > 
+                (cameraPosition - playerPosition).magnitude)
+                blockBehavior.SetIsFading(false);
+            else
+                blockBehavior.SetIsFading(true);
+        }
```

Comment:
Could use delegates to execute functions in FadeoutPlugins. Or could just directly set is fading in the FadeoutPlugin